ui-admin/src/study/participants/KitRequests.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useMemo, useState } from 'react'
+import { Enrollee, KitRequest } from 'api/api'
+import { StudyEnvContextT } from '../StudyEnvironmentRouter'
+import { ColumnDef, getCoreRowModel, useReactTable } from '@tanstack/react-table'
+import { basicTableLayout } from 'util/tableUtils'
+import RequestKitModal from './RequestKitModal'
+import { faPlus } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import {instantToDefaultString} from "../../util/timeUtils";
+
+function KitRequestAddress({ sentToAddressJson }: { sentToAddressJson: string }) {
+  const address = JSON.parse(sentToAddressJson)
+  return <div>
+    <div>{address.firstName} {address.lastName}</div>
+    <div>{address.street1}</div>
+    {address.street2 && <div>{address.street2}</div>}
+    <div>{address.city}, {address.state} {address.postalCode} {address.country}</div>
+  </div>
+}
+
+export default function KitRequests({ enrollee, studyEnvContext, onUpdate }:
+                                      {
+                                        enrollee: Enrollee,
+                                        studyEnvContext: StudyEnvContextT,
+                                        onUpdate: () => void
+                                      }) {
+  const [showRequestKitModal, setShowRequestKitModal] = useState(false)
+
+  const onSubmit = () => {
+    setShowRequestKitModal(false)
+    onUpdate()
+  }
+
+  const columns = useMemo<ColumnDef<KitRequest, string>[]>(() => [{
```

Comment:
declaring these columns outside the component function will save a useMemo call and make it clearer that the column defs don't change in response to the returned data

Reply:
I swear I saw somewhere that column definitions should be memoized, but maybe I was looking at the old v7 docs instead of the v8 docs. 🤦‍♂️

Reply:
it's correct that you have to memoize them *if* they are within the component function.  But these don't need to be declared within the component function, they can be an external const.